src/pages/billing/Project.js

```diff
@@ -221,10 +224,42 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
     _.map('workspace', workspaces)
   ), [billingProject, workspaces])
 
+  const spendReportKey = 'spend report'
+  const maybeLoadTotalCost = reportErrorAndRethrow('Unable to retrieve spend report data')(async () => {
+    if (!updatingTotalCost && totalCost === null && tab === spendReportKey) {
+      setUpdatingTotalCost(true)
+      const endDate = new Date().toISOString().slice(0, 10)
+      const startDate = new Date((Date.now() - spendReportLengthInDays * 24 * 60 * 60 * 1000)).toISOString().slice(0, 10)
+      const spend = await Ajax(signal).Billing.getSpendReport({ billingProjectName: billingProject.projectName, startDate, endDate })
+      const costFormatter = new Intl.NumberFormat(navigator.language, { style: 'currency', currency: spend.spendSummary.currency })
+      setTotalCost(costFormatter.format(spend.spendSummary.cost))
+      setUpdatingTotalCost(false)
```

Comment:
Since I am not changing `setUpdatingTotalCost` in a finally, if one gets an error from the API and want to try it again one must refresh the page. I think that is the behavior we want though, as we don't want repeated error attempts.